src/components/side_nav/side_nav.stories.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+
+import {
+  EuiSideNav,
+  EuiSideNavProps,
+  EuiSideNavHeadingProps,
+} from './side_nav';
+import { EuiIcon } from '../icon';
+
+const meta: Meta<EuiSideNavProps> = {
+  title: 'EuiSideNav',
+  component: EuiSideNav,
+};
+
+export default meta;
+type Story = StoryObj<EuiSideNavProps>;
+
+const componentDefaults: EuiSideNavProps = {
+  mobileBreakpoints: ['xs', 's'],
+  items: [],
+  // Aria-label and mobileTitle do not have defaults; they are being set here as they are shared between examples
+  'aria-label': 'Side navigation example',
+  mobileTitle: 'Mobile navigation header',
+};
+
+// Heading props shared across examples
+const _sharedHeadingProps: EuiSideNavHeadingProps = {
+  element: 'h1',
+  screenReaderOnly: false,
+};
+
+export const Playground: Story = {
+  args: {
+    ...componentDefaults,
+    heading: 'Elastic',
+    headingProps: _sharedHeadingProps,
+    isOpenOnMobile: false,
+    truncate: false,
+  },
+  render: ({ ...args }) => <StatefulSideNav {...args} />,
+};
+
+export const SideNavHeader: Story = {
+  args: {
+    ...componentDefaults,
+    items: [
+      {
+        name: 'Root item',
+        id: 'rootItem',
+        items: [
+          {
+            name: 'Child item',
+            id: 'childItem',
+            onClick: () => {},
+          },
+        ],
+      },
+    ],
+    heading: 'Navigation header',
+    headingProps: _sharedHeadingProps,
+  },
+  argTypes: {
+    // This story demos the header props; removing other props to prevent confusion
+    toggleOpenOnMobile: { table: { disable: true } },
+    isOpenOnMobile: { table: { disable: true } },
+    mobileBreakpoints: { table: { disable: true } },
+    items: { table: { disable: true } },
+    renderItem: { table: { disable: true } },
+    truncate: { table: { disable: true } },
+  },
+  render: ({ ...args }) => <EuiSideNav {...args} />,
+};
+
+const StatefulSideNav = (props: Partial<EuiSideNavProps>) => {
+  const [isSideNavOpenOnMobile, setIsSideNavOpenOnMobile] = useState(false);
```

Comment:
This should inherit from the storybook controls on initial mount/render
```suggestion
  const [isSideNavOpenOnMobile, setIsSideNavOpenOnMobile] = useState(props.isOpenOnMobile);
```